frontend-react/src/pages/submissions/SubmissionFilters.tsx

```diff
@@ -0,0 +1,119 @@
+ * table component contains the call and param passing to the API,
+ * and will use the context to get these values.
+ */
+function SubmissionFilters() {
+    const { updateFilter, clear, paginator } = useContext(
+        SubmissionFilterContext
+    );
+
+    /* Local state to hold values before pushing to context. Pushing to context
+     * will trigger a re-render due to the API call fetching new data. We have local
+     * state to hold these so updates don't render immediately after setting a filter */
+    const [localStartRange, setLocalLocalStartRange] = useState<string>();
+    const [localEndRange, setLocalEndRange] = useState<string>();
+
+    /* This workhorse function handles all Context changes with null checking */
+    const pushToContext = () => {
+        if (localStartRange && localEndRange) {
+            const srDate = new Date(localStartRange);
+            const erDate = new Date(localEndRange);
+
+            if (srDate < erDate) {
+                updateFilter(FilterName.START_RANGE, erDate.toISOString());
+                updateFilter(FilterName.END_RANGE, srDate.toISOString());
+            } else {
+                updateFilter(FilterName.START_RANGE, srDate.toISOString());
+                updateFilter(FilterName.END_RANGE, erDate.toISOString());
+            }
+        } else if (localStartRange && !localEndRange) {
+            updateFilter(
+                FilterName.START_RANGE,
+                new Date(localStartRange).toISOString()
+            );
+        } else if (localEndRange && !localStartRange) {
+            updateFilter(
+                FilterName.START_RANGE,
+                new Date(localEndRange).toISOString()
+            );
+        }
+    };
+
+    /* Pushes local state to context and resets cursor to page 1 */
+    const applyToContext = () => {
+        pushToContext();
+        if (paginator?.resetCursors) paginator.resetCursors();
+        if (paginator?.changeCursor) paginator.changeCursor(1);
+    };
+
+    /* Clears context and local state values */
+    const clearAll = () => {
+        // Clear context
+        if (clear) clear();
+
+        // Clear local state
+        setLocalLocalStartRange("");
+        setLocalEndRange("");
+    };
+
+    return (
+        <div data-testid="filter-container" className={StyleClass.CONTAINER}>
+            <div className={StyleClass.DATE_CONTAINER}>
+                <Label htmlFor="start-date" id="start-date-label">
+                    Submitted (Start Range)
+                </Label>
+                {/* BUG: Despite value being set to the local state, clearing it
+                     does not actually clear the displayed value. */}
+                <DatePicker
+                    id="start-date"
+                    name="start-date-picker"
+                    placeholder="Start Date"
+                    value={localStartRange}
+                    onChange={(val) => setLocalLocalStartRange(val)}
+                />
```

Comment:
Seems like `<DatePicker>` would use `rangeDate` property to limit dates that can be picked 

For example, does the current approach allows the user to select an end-date in the future? Or, can the end-date be BEFORE the start date?

Reply:
I've toyed with those props but to no avail; going to try again after demo prep!